Merge master

jfsiii opened the pull request:
## Summary

Merge in master to keep this feature branch current.

Created branch based on [comments from earlier PR](https://github.com/elastic/kibana/pull/38255#issuecomment-499704571)

```
git fetch upstream master
git checkout -b merge-master upstream/master 
git push origin merge-master
```

If I understand correctly, this is how to do additional "update" PRs
```
git checkout merge-master # switch to PR source branch
git pull upstream master # bring in latest upstream master
git push origin merge-master # create the PR from this branch
# open https://github.com/elastic/kibana/compare/feature-integrations-manager...jfsiii:merge-master?expand=1
```

This PR `merge-master` is from my fork. Other devs will create their own, or whatever they choose to call it, when they follow the steps shown in the initial block. The alternative I considered was top-level `elastic/kibana:merge-master`, but went with the less controversial approach to try this out.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- [ ] ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [ ] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [ ] ~~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
- [ ] ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] ~~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
- [ ] ~~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

dmlemeshko committed:
skip Code History functional tests on Firefox (#37954)

weltenwort committed:
[Logs UI] Include correct fields in "entries after" query (#37881)

This fixes #37704 by correcting the fields list used in the second part of the `getLogEntriesBetween` library function.

jgowdyelastic committed:
[ML] Reverting incorrectly removed scope parameters (#37873)

* [ML] Reverting incorrectly removed scope parameters

* removing  no-unused-vars comments

jgowdyelastic committed:
[Watcher] Reverting incorrectly removed scope parameters (#37875)

* [Watcher] Reverting incorrectly removed scope parameters

* removing  no-unused-vars comments

maryia-lapata committed:
[Vis: Default editor] EUIficate IP Ranges (#36896)

* Create IpRangeType and IpRanges controls

* Add validation

* Refactoring

* Add behavior when discarding changes

* Refactoring: create common input list

* Remove old template

* Move add btn to input_list, add placeholder

* Remove unused directives

* Remove unused translations

* Refactoring

* Use EuiButtonGroup instead of toggle button

* Update options ids, add aria-labels

* Remove unused translations

* Update mask model, update TS, update aria labels

* Add validation for CIRD mask

legrego committed:
Security - display message when logging out of kibana (#37234)

* display message when logging out of kibana

* update provider tests

* fix token logout test

* remove newline

legrego committed:
Feature Controls - update base privilege tooltip (#37868)

legrego committed:
Feature Controls - automatically grant access to short urls (#37532)

* automatically grant access to short urls

* adds API integration tests

* Update x-pack/test/api_integration/apis/short_urls/index.ts

Co-Authored-By: Brandon Kobel <brandon.kobel@gmail.com>

Bamieh committed:
[i18n] [docs] Update docs to encourage static i18n import (#37858)

* update documentation

* Update docs/development/plugin/development-plugin-localization.asciidoc

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/GUIDELINE.md

* Update packages/kbn-i18n/GUIDELINE.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/README.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

* Update packages/kbn-i18n/README.md

Co-Authored-By: gchaps <33642766+gchaps@users.noreply.github.com>

jkakavas committed:
Add section about OpenID Connect in docs (#37525)

justinkambic committed:
Add functional test to cover heartbeat location data. (#37891)

XavierM committed:
Fix bug when you delete column (#37946)

mattkime committed:
remove unneeded eslint-disable directives (#37922)

* remove unneeded eslint-disable directives

ryankeairns committed:
[Canvas] Fix workpad list overflow (#37345)

* fix workpad list overflow

* make file picker full width on small devices

* fix modal height, add comments

cjcenizal committed:
Add repository-azure autocompletion settings to Console (#37935)

Add repository-azure as an option when creating a repository.

Commit:
Add 1/2 second sleep to fix test (#37903)

spong committed:
[SIEM] Fix IP Overview query to query for most recent Host (#37723)

## Summary

As outlined in https://github.com/elastic/ingest-dev/issues/453, this PR fixes the `IP Overview` query to query for the most recent Host.

Now the Host Data shown will be the most recent host for the given IP regardless of whether `Source` or `Destination` is selected in the FlowTargetSelect.

![image](https://user-images.githubusercontent.com/2946766/58731558-46fb5200-83ac-11e9-97d5-7db5dd8ac84e.png)


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

Commit:
[SIEM] Host Details Page Updates (#37769)

* toggle to exclude host ID links on host details pg

* first attempt at hiding host column

* switch to conditionally splicing array of columns

* exclude dest from auth table on host details pg

* minor overview style touchups

* rm some extranous EuiFlexGroup; more need removal

* snapshots update

* Revert "snapshots update"

This reverts commit bdd6b8e3165b7488ade99a349d1a181f4abd17f9.

* update siem snapshots

* remove unused arg

* change uncommon process column order per #503

* simplify if condition, per steph

* update tests, per steph

* simplify `hostIdRenderer` per xavier

* update snapshots

* fix/improve unit tests, per xavier

* Revert "update snapshots"

This reverts commit dc0f79606bb508c657cf380eaa9028f1c3e0c845.

* add host links in uncommon processes table

* update uncommon process test, thx to xavier

FrankHassanabad committed:
[SIEM] Authentication fixes for winlogbeat and ordering (#37718)

## Summary
  * Enabled winlogbeat to work with workaround for authentication_failed (until that is fixed)
  * Fixed ordering to be successes first, failures second
  * https://github.com/elastic/ingest-dev/issues/491
  * https://github.com/elastic/ingest-dev/issues/469

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

zfy0701 committed:
[Code]: add ui.enabled flag and fix uneccessary usage of enabled flag (#38018)

chandlerprall committed:
Make plugins/reporting config snapshot deterministic (#38019)

cqliu1 committed:
[Canvas] Fixes broken elements in web logs sample data workpad (#37931)

* Fixed broken elements in logs sample workpad. Cleaned up expression formatting in each element

* Removed return characters at the beginning of expressions

kobelb committed:
Fixing isMigrated bug, when migrations failed we incorrectly ignored it. (#37674)

Prior to this change, when migrations failed (for example when a
.kibana_1 index already exists and no alias does) the awaitMigration
promise would resolve incorrectly. This allowed subsequent plugins,
including Spaces, to proceed using the SavedObjectsClient/Repository
to create documents. This commonly caused a .kibana index to be created,
which would then block the creation of a .kibana alias, and everything
would grind to a halt.

mw-ding committed:
[Code] update percentage (#38003)

mw-ding committed:
[Code] Filter out binary files for indexer (#37945)

lukeelmers committed:
Fix bad type import from interpreter. (#37741)

mw-ding committed:
[Code] use empty placeholder when search filters returns empty results (#38006)

igoristic committed:
Optimized content to be more vertical for resizing consistancy (#37120)

* Optimized content to be more vertical for resizing consistancy

* Fixed snapshots

* Feedback changes

* Update _summary_status.scss

* Fixed column order for testing

Commit:
[Code] fix hover in monaco-editor 0.17 (#37964)

zfy0701 committed:
[Code]: Use more robust way to handle url (#38060)

Commit:
[Code] fix url and encoding for java langserver on Windows (#38064)

Commit:
fix(code/frontend): catch structure generating error in saga (#37957)

flash1293 committed:
Add error handling to expression runner (#37968)

Commit:
fix(code/frontend): improve regexp that matches the container of a symbol (#37856)

Commit:
fix(code/frontend): should not polling status if import repo not succeed (#37955)

simitt committed:
[apm] update index pattern (#37866)

related to https://github.com/elastic/apm-server/pull/2239

peteharverson committed:
[ML] Fixes loading of Single Metric Viewer if partition field is text (#37975)

* [ML] Fixes loading of Single Metric Viewer if partition field is text

* [ML] Fix failing test and edit translations for cardinality check

* [ML] Edits to fields_service.js following review

mikecote committed:
Fix export saved objects API docs (#37991)

mikecote committed:
Update import saved objects API docs to use curl examples (#37997)

* Remove copy as curl

* Update import docs to use curl examples

justinkambic committed:
Remove deprecated feature from elasticsearch queries. (#37995)

justinkambic committed:
[Uptime] [Bug] Disable integration links for apps that are not present (#37284)

* Add tests and checks for disabled/enabled integration link.

* Add empty state if all integrations are unavailable.

* Fix broken type.

jgowdyelastic committed:
[ML] Fix for constant polling of deleting job tasks (#38013)

* [ML] Fix for constant polling of deleting job tasks

* fixing comment

mikecote committed:
Fix URLs from displaying too long in saved objects management UI (#37987)

* Fix URLs from displaying too long

* Add encodeURIComponent

kobelb committed:
Updating mixin-deep to 1.3.1 (#38057)

mistic committed:
Relative cache loader (#36131)

* chore(NA): first efforts to enable cache-loader relative paths and disable babel-register cache. chore(NA): allow to use cache-loader results everywhere as a persistent cache.

* chore(NA): last change to support cache loader across envs.

* fix(NA): cache-loader config generator function scope.

* refact(NA): static cache config for dll compiler.

* chore(NA): support for force writable cache flag. use built assets for cache on distributable.

* chore(NA): update cache-loader to last version. fix(NA): logic bug with force writable cache flag.

* chore(NA): move optimize cache to built_assets.

* chore(NA): relative paths for sourcemap configurations.

* chore(NA): remove redudant check for process.env var.

* refact(NA): simplify double negated expression.

* chore(NA): remove webpack prefix from devtoolModuleFilenameTemplate.

* chore(NA): rename cache write force flag to KBN_CACHE_LOADER_WRITABLE.

* chore(NA): readd babel register cache.

* chore(NA): change .cache to .ui_bundles_cache

* chore(NA): change .ui_bundles_cache to cache/ui_bundles

* chore(NA): update gitignore with .babel_register_cache.json

* chore(NA): rollback devtool module filename template to asbolute resource path.

* chore(NA): apply cache-loader patch.

clintandrewhall committed:
[Canvas] Interpreter type conversion (#37300)

* Move Expression types to OSS.

* Add interfaces for KibanaContext and KibanaDatatable types.

* Convert esaggs expression function to TypeScript.

* Clean up KibanaDatatable and KibanaContext types.

* Convert functions to use new Interpreter types

* Update ArgumentString-related types to simply be TypeString.

* Rename types.ts to index.ts so types can be imported at the top level.

* Update ArgumentString-related types to simply be TypeString.

* Rename types.ts to index.ts so types can be imported at the top level.

* Remove unused code from arguments types.

* Convert x-pack imports of OSS plugins from relative to absolute.

* Fix esaggs import.

* Addressing self feedback

ryankeairns committed:
default shape aspect ratio to false (#38030)

dmlemeshko committed:
functional tests/firefox wait for loading after page refresh (#38082)

nickpeihl committed:
Use production EMS manifest (#38050)

vitalics committed:
[TSVB] [AT] implement new test cases for time series formatter (#37506)

* implement new test cases for time series formatter

justinkambic committed:
Remove "common" options from SDP that have a 0ms difference between end and start time. (#37395)

lukeelmers committed:
Fix broken filtering on metric visualizations. (#38029)

Bamieh committed:
Allow headers config to propagate in `callAsCurrentUser` (#34610)

* add telemetry query source when hitting monitoring indexes

* allow headers to pass through

* Update src/core/server/elasticsearch/scoped_cluster_client.ts

* fixes from bill code review

* split test cases

* fix functional test

* check if issue from intersect function

* more debug

* check for duplicate values instead

* code review updates

* remove mockClear

nreese committed:
[Maps] replace FormattedText with FormattedMessage (#38016)

* [Maps] replace FormattedText with FormattedMessage

* fix Capitalization

Commit:
Move shimming of workpad CSS to only happen on fetch (#38117)

Commit:
Fix discover not using an index pattern when 2 are created (#38053)

zfy0701 committed:
[Code]: unassociate html with typescript langserver (#38150)

cjcenizal committed:
Skip Index Manamgent API integration tests that are inapplicable to Cloud, when they are run on Cloud. (#38045)

kobelb committed:
Updating stringstream to 0.0.6 (#38155)

lukeelmers committed:
ui settings mocks: mock observables' return values. (#37584)

joshdover committed:
[new-platform] Simplify the frontend integration with the legacy platform (#37734)

TinaHeiligers committed:
KQL in TSVB (#36784)

* Squashed commit of the following:

commit 0cff8243a99ff23421c3cbb15d62b26332a31e21
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Mon May 20 16:16:15 2019 -0700

    Adds ignore_lobal_filter YesNo component back

    Deletes fixtures file

commit e54e1690c9c46e2de5e747d14030a7e27e74c138
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Mon May 20 14:39:21 2019 -0700

    squash-merges adding KQL with the full QueryBar and SavedObject migrations

Debugging

Adds filter defaults

Updates index pattern for the QueryBar on componentDidUpdate

squash-merges adding KQL with the full QueryBar and SavedObject migrations required

Adds ignore_lobal_filter YesNo component back

Deletes fixtures file

Updates migration test number

* Changes QueryBar to QueryBarInput in TimeseriesPanelConfigUi

Replaces QueryBar with QueryBarInput in Timeseries

Changes QueryBar to QueryBarInput in TSVB Metric

Changes QueryBar to QueryBarInput in other TSVB visualisation editors

Removes async from componentWillMount

changes indexPattern to indexPatternObject in series and table query request processors

* Squashed commit of the following:

commit 2c3107963cff20b2d24323fddfaf80c848b86848
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 15:15:06 2019 -0700

    Cleans up comments

commit 943150d4b9b71183e9d5b26c95449ea4519aad4c
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 14:09:53 2019 -0700

    Cleans up parser check more

commit d31de88525595720556440ae71515f5772774eaa
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 14:01:02 2019 -0700

    Cleans up parser check

commit b592071b1d939ec2df9b9c921d57a22dac54e703
Author: Christiane Heiligers <christiane.heiligers@elastic.co>
Date:   Thu May 23 13:16:55 2019 -0700

    skips server call on invalid kuery queries

Cleans up

Deletes duplicate migration function

Adds 's' to split_by_filters method

Resolves issues after rebase

* Removes quotes around index pattern strings passed into fetch_index_patterns, adds comments on reasoning

* queryBar to queryBarInput

Adds transformFilterStringToQueryObject back to visualizations migrations that was lost during rebase

* Deletes unused code

* Fixes migrations test

* compiles visualizations migrations for version 7.3.0

* temporary change to query_bar_input fetch_index_patterns file

formatting

Includes code from PR#37413

Clean up

* Formatting

* Uses series index pattern when override_index_pattern is selected and an index pattern is provided

* Removes empty object as alternative to localStorage

* Removes handelQueryChange from where it is not needed

* extracts retrieval of the default query language from uiSettings into a helper method

* Resolves some PR comments

* Handles Query Error from an invalid syntax query

* Converts string queries into objects with kuery as the language in saved_objects data for ecommerce, flights and logs TSVB visualizations

* Deletes unused translation items

* Adds optional id prop to the query_bar_input and makes the screenTitle optional

* Updates [eCommerce] Sold Products per Day

* Removes screenTitle from component QueryBarInput

* Wraps index pattern strings in a double quote to allow for names such as 'logstash-*'

mw-ding committed:
[Code] search functional test to cover file typeahead query case insensitive (#38012)

mw-ding committed:
[Code] Improve the document search highlights for non-english query (#38154)

jen-huang committed:
Fix i18n issues (#38169)

MichaelMarcialis committed:
[SIEM] Beta badge on overview page (#38161)

* add beta badge to overview header

* update test

* updates snapshot

nickpeihl committed:
Fix sample data functional test (#38184)

Commit:
[Code] fix another path problem on windows (#38072)

stephmilovic committed:
[SIEM] Additional Overview Network & Hosts metrics (#38005)

[SIEM] Additional Overview Network & Hosts metrics (#38005)

mistic committed:
chore(NA): declare kbn-babel-preset as peer dependency for kbn plugin helpers. (#37710)

clintandrewhall committed:
[Canvas] Localize Function Errors (#37381)

* [Canvas] Localize Function Errors

* Update test

* Fix enum usage

Commit:
[Code] improve setup guide (#38218)

FrankHassanabad committed:
[SIEM] Changes out event.severity for message in timeline and fixes sorting bug (#38158)

## Summary

  * Change out event.severity for message in timeline
  * Fixed critical crash where you should not be able to sort on timeline columns that are not aggregatable
  * https://github.com/elastic/ingest-dev/issues/513
  * https://github.com/elastic/ingest-dev/issues/496

Error toaster fixes:
<img width="348" alt="Screen Shot 2019-06-05 at 1 59 01 PM" src="https://user-images.githubusercontent.com/1151048/58986428-97065a00-879a-11e9-8ff9-003ac29f6d9e.png">

What the timeline looks like with severity replaced with message:
<img width="883" alt="Screen Shot 2019-06-05 at 1 55 19 PM" src="https://user-images.githubusercontent.com/1151048/58986459-a71e3980-879a-11e9-8144-f16f055bd53d.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

spong committed:
[SIEM] Removing draggability from message field (#38166)

## Summary

Currently, searching the `message` field within Timeline will not function properly, so in effort to prevent the user from searching with it, this PR removes the areas where `message` is draggable, which includes:

* Host Events Table
* Host Details Events Table
* Timeline Renderers
* Timeline Event Details

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

andrew-goldstein committed:
## [SIEM] Disables Timeline Provider Mutations While Loading Data (#38185)

## Summary

* Fix: disables dropping a new data provider provider on the timeline, deleting an existing data provider, or choosing an action from the data provider context menu while the timeline is loading data to prevent draggable errors, like the one pictured in the screenshot below:

![draggable-error](https://user-images.githubusercontent.com/4459398/58996789-ebb7ce00-87b6-11e9-9490-47a0df6f89ab.png)

* Fix: the timeline settings icon appears too close to the `Refresh` button:
### Before (Chrome)

![timeline-gear-before](https://user-images.githubusercontent.com/4459398/58997030-e6a74e80-87b7-11e9-86df-cb67592e63a7.png)

### After (Chrome)

<img width="1679" alt="timeline-gear-after-chrome" src="https://user-images.githubusercontent.com/4459398/58997064-1ce4ce00-87b8-11e9-9ffb-0f412e0018ad.png">

### After (Firefox)

<img width="1672" alt="timeline-gear-after-firefox" src="https://user-images.githubusercontent.com/4459398/58997113-4f8ec680-87b8-11e9-9b52-dc8a13d51361.png">

### After (Safari)

<img width="1680" alt="timeline-gear-after-safari" src="https://user-images.githubusercontent.com/4459398/58997169-911f7180-87b8-11e9-8186-0ff31e41ad53.png">

### After (IE 11)

<img width="1354" alt="timeline-gear-after-ie11" src="https://user-images.githubusercontent.com/4459398/58997306-40f4df00-87b9-11e9-844d-b12c7e16f7cb.png">

* Fix: the `Open Timeline` modal is showing bulk-delete actions

### Before (Chrome)

![open-timeline-before](https://user-images.githubusercontent.com/4459398/58997349-700b5080-87b9-11e9-8299-7dbeccbd8ff9.png)

### After (Chrome)

<img width="1008" alt="open-timeline-after" src="https://user-images.githubusercontent.com/4459398/58997381-9af5a480-87b9-11e9-9909-2d68bd5ff2c3.png">

* Fix: action icons on the `Timelines` page and `Open Timelines` modal are not aligned with the counts

### Before (Chrome)

<img width="1552" alt="all-timelines-before" src="https://user-images.githubusercontent.com/4459398/58997554-659d8680-87ba-11e9-868c-d92ad72f44b8.png">

### After (Chrome)

<img width="1550" alt="all-timelines-after" src="https://user-images.githubusercontent.com/4459398/58997656-d8a6fd00-87ba-11e9-9038-c0ca464f8c4d.png">

* Fix: Changed the title text of the fields browser from `Select Fields` to `Customize Columns`

### Before (Chrome)

<img width="910" alt="fields-browser-before" src="https://user-images.githubusercontent.com/4459398/58997616-a4334100-87ba-11e9-9cd8-02ed9d265bf2.png">

### After (Chrome)

<img width="910" alt="fields-browser-after" src="https://user-images.githubusercontent.com/4459398/58997691-fa07e900-87ba-11e9-8cd9-6412d2cef8f0.png">

https://github.com/elastic/ingest-dev/issues/471

ogupte committed:
[APM] Moves the APM index creation from server startup (#37965)

* [APM] Closes #37499 by moving the APM index creation from server startup
to savedObject request for the index pattern. It first checks if the
index pattern is saved, if not it creates it.

* [APM] provide more meaninful status codes in the default error handler

rudolf committed:
[Core] Rewrite saved objects in typescript (#36829)

* Convert simple files to TS

* Fix jest tests

* Rename saved_objects_client{.js => .ts}

* WIP saved_objects_client

* saved_objects repository{.js => .ts}

* includedFields support string[] for type paramater

* Repository/saved_objects_client -> TS

* Fix tests and dependencies

* Fix saved objects type errors and simplify

* saved_objects/index saved_objects/service/index -> ts

* Fix saved objects export test after switching to typed mock

* Workaround type error

* Revert "Workaround type error"

This reverts commit de3252267eb2e6bf56a5584d271b55a7afdc1c53.

* Correctly type Server.savedObjects.SaveObjectsClient constructor

* saved_objects/service/lib/index.{js -> ts}

* saved_objects/service/lib/scoped_client_provider{js -> ts}

* Typescriptify scoped_client_provider

* Fix x-pack jest imports

* Add lodash/internal/toPath typings to xpath

* Introduce SavedObjectsClientContract

We need a way to specify that injected dependencies should adhere to the
SavedObjectsClient "contract". We can't use the SavedObjectsClient class
itself since it contains the private _repository property which in TS is
included in the type signature of a class.

* Cleanup and simplify types

* Fix repository#delete should return {}

* Add SavedObjects repository test for uncovered bug

Test for a bug in our previous js implementation that can lead to data
corruption and data loss.

If a bulkGet request is made where one of the objects to fetch is of a type
that isn't allowed, the returned result will include documents which have the
incorrect id and type assigned. E.g. the data of an object with id '1' is
returned with id '2'. Saving '2' will incorrectly override it's data with that
of the data of object '1'.

* SavedObject.updated_at: string and unify saved_object / serializer types

* Cleanup

* Address code review feedback

* Don't mock errors helpers in SavedObjectsClient Mock

* Address CR feedback

* CR Feedback #2

* Add kibana-platform as code owners of Saved Objects

* Better typings for SavedObjectsClient.errors

* Use unknown as default for generic type request paramater

* Bump @types/elasticsearch

* Fix types for isForbiddenError

* Bump x-pack @types/elasticsearch

XavierM committed:
fix math (#38132)

* fix math

* add unit test

* review I

Commit:
Typescript dashboard app code. (#37527)

* Typescript dashboard app code. Pulls out part of the massive embeddable API PR.

I typscripted this code as part of that PR because it helped me find errors, but really unnecessarily blew up the size of that PR. So pulling out.

* User filter types from kbn-es-query

* Address code review feedback

* Replace require with import

mshustov committed:
Route tags (#37344)

* expose route info in KibanaRequest

* update mocks in test

* make tags readonly, getRouteInfo is private method

* add mocks for hapi internals

* mode deepFreeze to core utils level as it env agnostic

* freeze route props

* fix typo

* add tests for route options

* fix integration tests. deep_freeze was moved under core utils

* add comments, expose public types and regenerate docs

* address comment. remove unnecessary async in route handlers

* make routeSchema optional instead of union with undefined

* @skaapgif improvements

* update docs

peteharverson committed:
[ML] Close job after forecast errors if starting from closed (#38253)

* [ML] Close job after forecast errors if starting from closed

* [ML] Pass closeJob flag to error handler if forecast ID undefined

peteharverson committed:
[ML] Only add ML links for sample data sets if full license (#38120)

* [ML] Only add ML links for sample data sets if full license

* [ML] Update following review

kobelb committed:
[docs][skip ci] Read only access for @elastic/kibana-app applications  (#35560)

* Adding read-only mode section to Discover

* No more "mode" or "badge" terminology

* Putting newline back

* Adding "Saving" section for visualize

* Adding dashboard read only access section

* Timelion gets read-only documentation

* Adding read only access section to index patterns and advanced settings

* Adding graph read only access section

alvarezmelissa87 committed:
revert dragselect version (#38183)

streamich committed:
fix: 🐛 stop showing spinner only on error (#38289)

Commit:
[dev/license_checker][dev/npm] reactor, ts-ify, de-grunt (#37807)

Commit:
[dev/failed_tests] try to ignore irrelevant failures (#37902)

* [dev/failed_tests] try to ignore irrelevant failures

* fix typo

kertal committed:
Undo setting of $scope.refreshInterval at dashboard (#37175)

* Add test for passing on dashboard url params to timepicker values

* Revert unnecessary setting of $scope.refreshInterval

refreshInterval is set at $scope.model, no need to set it directly at $scope like in discover and vis

kobelb committed:
Updating tar to 2.2.2 (#38182)

kobelb committed:
Updating underscore.string to 3.3.5 (#38180)

kobelb committed:
Updating axios to 0.18.1 (#38179)

kobelb committed:
Updating js-yaml to 3.13.1 for more dependencies (#38160)

Renovate recently updated js-yaml in
https://github.com/elastic/kibana/pull/37809. However, a few
indirect dependencies weren't upgraded to use this new version.

simianhacker committed:
[Infra UI] Fixes #37999 - Reset error when IP address changes to valid entry (#38022)

* [Infra UI] Fixes #37999 - Reset error when IP address changes to something valid

* movnig the setError above try/catch

nreese committed:
[Maps] fix tooltip text overlap and text overflow (#38271)

* [Maps] Use table to display tooltip properties

* Design cleanup

- Added dropshadow
- Cleaned SASS
- Removed unnecessary wrappers around close button and centered it
- Change loading state to be centered
- Fixed z-index so tooltip is on top of controls
- Condensed padding/spacing

* Fix overlays in IE

* revert yarn.lock changes

stephmilovic committed:
[SIEM] fix ipv6 bug and write tests for ip details page (#37528)

fix ipv6 bug and write tests for ip details page

zfy0701 committed:
[Code]: remove hard code node heapsize (#38304)

Commit:
[ci] disable firefox tests

Commit:
update yarn.lock

tsg committed:
[SIEM] Update links to docs and website (#38313)

* Update links to docs and website in the SIEM app

* Update also link text

* Capitalize solution

nickpeihl committed:
Add link to 7.2 ems landing page (#38284)

FrankHassanabad committed:
Fixed cyclic dependencies (#38321)

## Summary

Fixes cyclic dependencies

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

martinliu committed:
Add language codes in comment (#37635)

It would be good to see a list here, as long as we will add more languages. 
A user will not have to look into any other document to find a language code.
That will be more easier for any one who configure this file.

Commit:
[jest] don't be crazy, run with caching enabled (#38332)

spong committed:
[SIEM] Fix overlapping timestamp tooltip in expanded Event Details (#38328)

## Summary

Fixes issue where multiple tooltips are shown on `date` and `duration` fields within the expanded Event Details UI (https://github.com/elastic/ingest-dev/issues/481).

### Before:
<img src="https://user-images.githubusercontent.com/2946766/59068176-b8397a00-8870-11e9-9b04-ee16d89ac7d4.png" width=400>

<img src="https://user-images.githubusercontent.com/2946766/59068283-00f13300-8871-11e9-989d-986e9ae00e2a.png" width=400>


### After:
<img src="https://user-images.githubusercontent.com/2946766/59068223-d901cf80-8870-11e9-9a28-1322adcfe570.png" width=400>

<img src="https://user-images.githubusercontent.com/2946766/59068321-15353000-8871-11e9-8e33-2ffaa0b0c9c2.png" width=400>

<img src="https://user-images.githubusercontent.com/2946766/59068621-d653aa00-8871-11e9-8803-8be01a20e585.png" width=400>



### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~
- [ ] ~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- [ ] ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- [ ] ~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~
- [ ] ~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] ~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~
- [ ] ~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~

spong committed:
Fixing crash when collapsing a second event details (#38303)

mw-ding committed:
[Code] adjust clone progress and not found rendering in source view page (#38213)

renovate[bot] committed:
Update dependency history to v4.9.0 (#37799)

* Update dependency history to v4.9.0

* importing submodules is deprecated, import root package

* chore: bump history-extra to 5.0.1

compatible with history@^4.9.0

Commit:
re-enable yarn.lock validation checks

Commit:
[scripts/update_prs] implement script to merge base branch into prs (#38302)

Commit:
re-enable yarn.lock validation checks

Commit:
update kbn-pm dist

renovate[bot] committed:
Update dependency @microsoft/api-extractor to v7.1.6 (#37760)

* Update dependency @microsoft/api-extractor to v7.1.6

* regenerate docs

* OverlayService: Import react instead of relying on global

nreese committed:
[Maps] ensure labels extracted from WMS capabilities are unique (#38311)

* [Maps] ensure labels extracted from WMS capabilities are unique

* clean up

* clean up

Commit:
[Code] change the port number before respawn a new lang-server process (#38090)

angorayc committed:
[SIEM] Kpi network enhancement (#36915)

* add histogram

* clean up with hooks

* fix types and inport

* update mock data

* fix barchart

* fix integration test

* wrapping hooks with functional component

* update snapshot

* remove active agents and network events histogram from kpi network

* amend chart height

* fix integration test

* update wording

* add readonly type

* add references

* update translations and add constants

* clean up types for kpi host

* clean up types for kpinetwork

* remove a redundant type

* fix tests

* replace react memo with recompose pure

* add unit test and update mock data

* remove redundant type

renovate[bot] committed:
Update webpack related packages (major) (#37552)

* Update webpack related packages

* raw-loader@2 uses esm, so we must import

Commit:
Revert "[Code] change the port number before respawn a new lang-server process (#38090)"

This reverts commit 05947fed0c367403fff0f5a6be68682e7bb8bc68.

renovate[bot] committed:
Update dependency leaflet to v1.5.1 (#37811)

* Update dependency leaflet to v1.5.1

* update baseline image

* Fix map draw controls

* Update CSS overrides to match updated package

renovate[bot] committed:
Update dependency pdfmake to v0.1.56 (#37887)

renovate[bot] committed:
Update dependency pegjs to v0.10.0 (#37889)

* Update dependency pegjs to v0.10.0

* PEG.buildParser() => PEG.generate()

renovate[bot] committed:
Update dependency pngjs to v3.4.0 (#37890)

renovate[bot] committed:
Update dependency @microsoft/api-extractor to v7.1.8 (#38364)

ppisljar committed:
removing provider (#38221)

ppisljar committed:
removing provider (#38223)

mshustov committed:
update docs how to add typs for config (#38224)

mshustov committed:
Session storage refactoring (#37992)

* Kibana request keep a reference to raw request. used to bind hapi-cookie

* CookieSessionStorage should work with KibanaRequest

as soon as registerAuth refactored to restrict access to hapi Request, CookieSessionStorage won't work with hapi request directly

* change registerAuth public api

* adopt auth lifecycle tests

* move lifecycle auth tests from integration to unit and adopt to new api.

* mark toRawRequest as internal to prevent exposure

* generate docs

* reword test cases

* mark Request internals in tsdoc

timroes committed:
Fix broken Vega context menu (#38095)

* Fix broken Vega context menu

* Remove debug output

* Improve code comment

Commit:
Convert Markdown components to TS (#38081)

* Convert Markdown components to TS

* Fix jest snapshots

nreese committed:
Fetch options for input control child select boxes (#38148)

renovate[bot] committed:
Update dependency geckodriver to v1.16.2 (#37798)

renovate[bot] committed:
Update webpack related packages (major) (#38372)

renovate[bot] committed:
Update dependency proxyquire to v1.8.0 (#37913)

ppisljar committed:
Vislib private removal part2 (#37850)

justinkambic committed:
[Uptime] [Bug] Support KQL "OR" filters (#37564)

* Fix typo in variable name.

* Add support for kql-rendered output from filter bar.

* Support nested status query output.

* Expand complicated mock data.

* Add comments, clean up code.

* Improve naming, comments, simplify function bodies.

renovate[bot] committed:
Update dependency nock to v10.0.6 (#37886)

mshustov committed:
Document mock convention (#37888)

* add docs how to use config in NP

* do not encourage storing initializer context in plugin

* add mock convention docs

* address comments

* address Josh comments

Commit:
disable flaky test

mshustov committed:
disable renovate for developer doc generators. (#38239)

* disable renovate for doc generators. we have to regenerate docs on a version update

* use packageNames for static values

Commit:
Switch from old error messages to toast notifications (#38116)

* Replace notifier.error by toast notifications

* Fix incomplete mock

* Improve pipeline error title

Commit:
Fix typo in failure filtering (#38374)

* include a default value for failure string

* test the failure filtering

* oops, reverted too many changes

simianhacker committed:
[Infra UI] Fixes Uptime link to display for pod and containers (#38125)

* Fixes uptime link to display for pod and containers

* Making code more readable

* Another round of readability changes

* removing extra truthy check

simianhacker committed:
[Infra UI] Fixing Metrics Explorer chart title overlapping (#38059)

* [Infra UI] Fixing Metrics Explorer chart title overlapping

* Adding tooltip to title

Commit:
remove accidentally committed stats file from repo

Bargs committed:
Enable use of KQL and autocomplete in filters agg editor (#37287)

This PR updates the filters agg editor to use the full QueryBar component, enabling use of KQL and autocomplete inside the editor for this aggregation in Visualize.

ppisljar committed:
Vislib private removal part3 (#37852)

* removing Private

daveyholler committed:
Fixing padding issues in the toolbar (#38315)

daveyholler committed:
Cursor on symbol hover. (#38062)

Commit:
Reapply custom element telemetry (#37113)

* Revert "Revert "[Canvas] Adds telemetry info for custom elements (#36177)" (#37100)"

This reverts commit 3f4c2c89ecba9aeb6e77b88044b39cd2b371dcbd.

* Cast fromExpression unknown type

* PR Feedback

justinkambic committed:
[Uptime] [Bug] Fix spaces in filters (#38320)

* Fix typo in variable name.

* Add support for kql-rendered output from filter bar.

* Support nested status query output.

* Expand complicated mock data.

* Add comments, clean up code.

* Improve naming, comments, simplify function bodies.

* Refactor filter selection method to handle arbitrary input that is unrelated to status.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/964/)

Commit:
Embeddable API V2 (#37510)

* Embeddable API plugin

* Expose new embeddableActions uiExport endpoint

* Add missing getInjectedUIAppVars to Server type

* Add jest tests

* Most basic skeleton of the sample plugin to see if it still kills ci

* strip even more out to see if it passes ci

* It passed, put back init fn

* add back in a uiExport to see if that is what is killing ci

* Passed again, add back embeddableActions uiExports and require kibana line

* Add everything back but the sass import

* Found two bugs with customize panel title action, added jest test coverage and fixes

* Functionally test embeddable explorer plugin

* Addressing review feedback part 1

* Simplify action context menu - remove support for nested actions/child panels, which was never exposed anyway.

* More review feedback

* Spread out orders to let developers inject their own actions in the middle.

* Remove check for overwriting ApplyFilterAction

* use createRegistry for EmbeddableFactoryRegistry

* Add comment for getInheritedInput

* Use kbn-es-query Filter types

* Fix missed file after createRegistry switch over for EmbeddableFactoryRegistry

* Use delete instead of setting to undefined

* upgrade EUI to match kibana version

* Add getIsContainer on base embeddable class

* Run functional tests of sample plugin

* move all tests classes outside __test__ and inside test_samples folder so not to potentially create issues with mocha

* fix: 🐛 remove unused imports

* fix: 🐛 set to undefined explicitly

* chore: refactored getRoot functionality

* fix: refactor miss from folder rename __test__ -> test_samples

* fix: add eui switch to let the user indicate “hide this title” to improve ux.

* fix: customize panel flout test

* Try to fix issue with multiple rxjs bundles I hit before.

See https://github.com/ReactiveX/rxjs/issues/3828

* Use an rxjs polyfill to work around the issue of multiple rxjs bundles

* chore: change customize panel flyout to modal post design feedback

* capitalize Reset

* fix: type errors in customize panel modal

* fix: remove extra line added to prevent EUICallout overflow from spilling over panel edges

The bug this was intended to fix doesn’t appear yet because dashboard isn’t using this, and it causes other issues, so for now, just remove.

* Clean up some classes and SASS

* Inline styles needed to get the containers to take up the full height of the app.

* Use the same old style of error messaging in panels

* chore: add untilEmbeddableLoaded tests, expose on interface, and use in embeddable_child_panel

* Remove unused styles, add padding

* Verify trigger Context containers an array of Filters for apply filter action

* Remove panel.embeddableId param, use panel.explicitInput.id instead

* Check `isCompatible` as well as the type guard needed for typescript.

* executeTriggerActions should account for getHref

* Remove unnecessary check for context being defined

* use npStart instead of getNewPlatform

* Conform to latest NP changes

* Use new np_mocks in all the tests

* Address review feedback on rxjs polyfill

kobelb committed:
Updating chownr to 1.1.1 (#38177)

gchaps committed:
[DOCS] Adds docs for snapshot repositories UI (#38159)

* [DOCS] Adds docs for snapshot repositories UI

* [DOCS] Incorporates review comments into snapshot docs

renovate[bot] committed:
Update dependency apollo-cache-inmemory to v1.6.2 (#38435)

renovate[bot] committed:
Update dependency puid to v1.0.7 (#38436)

renovate[bot] committed:
Update dependency regression to v2.0.1 (#38437)

renovate[bot] committed:
Update dependency rimraf to v2.6.3 (#38438)

mw-ding committed:
[Code] better limit the size of indexed files (#38327)

* [Code] better limit the size of indexed files

* [Code] separate the batch index helper for document and symbols

legrego committed:
Security - remove usage of legacy notifier (#38260)

* remove usage of legacy notifier

* fix superfluous whitespace changes

* fix redirection on role management screen

* extract session expiration warning into its own component

igoristic committed:
Issue 28875:  Using i18n instead of injectI18n (#37845)

* Using i18n instead of injectI18n

* Update x-pack/plugins/monitoring/public/components/apm/instance/status.js

Co-Authored-By: Ahmad Bamieh <ahmadbamieh@gmail.com>

* Update x-pack/plugins/monitoring/public/components/table/table.js

Co-Authored-By: Maryia Lapata <mary.lopato@gmail.com>

* Addressed PR code review feedback

igoristic committed:
Issue 27126: Added a link to License Management (#37771)

* Added a link to License Management

* Added textAlign to EuiText instead

Commit:
Implement renovate config generator (#38369)

* reformat renovate.json5 with json5 cli, add header

* add scripts for rengerating renovate config from ts files

* regenerate renovate config

* verify renovate.json5 is up to date in CI

* add description to the build_renovate_config script

* auto-group types packages

* fix typo

* mention version pinning in doc block

* support disabling renovate for package groups

* scan all package paths for @-types deps

* regenerate renovate config

* regenerate renovate config

* name scoped package regexp

* move some of the new logic out of package_groups.ts

* flatMap in one pass

FrankHassanabad committed:
[SIEM] Add build step to find cyclic deps in SIEM project (#38329)

## Summary

* Adds a build step to find cyclic deps and error out in the SIEM project when found as part of a build step.

* Fixes typescript files that have cycles with their constants and types.

You run this like so:

```
cd kibana/x-pack/plugins/siem
node scripts/check_circular_deps.js
```

And get this message:
<img width="846" alt="Screen Shot 2019-06-07 at 11 22 24 AM" src="https://user-images.githubusercontent.com/1151048/59121970-b629f680-8916-11e9-8683-326a881c4725.png">

Or you will get an error with a listing of deps that need to be updated which have cycles in them.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

renovate[bot] committed:
Update dependency simple-git to v1.113.0 (#38456)

renovate[bot] committed:
Update dependency rsync to v0.6.1 (#38454)

cqliu1 committed:
[Canvas] Clean up expression function definitions (#37305)

* Cleaned up common function defs

    fixed broken tests

    Remove boolean and null as context/arg valid types in lt, lte, gt, and gte

    Added 'function' alias

    Updated comparator tests

    Added neq to options in compare function

    null audit

* Deleted obsolete 'browser' and 'server' functions

* More clean up

* Removed server and browser function i18n strings

* Fixed broken tests

* Addressing feedback

* Removed browser and server from function_help

* Added ts-ignore for constant import

* Fixed help text label

* Fixed font

simianhacker committed:
[Infra UI] Link to node detail page from Metrics Explorer (#37136)

* [Infra UI] Link to Metrics Detail page from Metrics Explorer for compatible group bys

* Adding test for incompatible group bys

* renaming for clearity

* Changing out SourceQuery for SourceConfiguration

* Fixing tests to ensure link it validated

* Make tests more robust

* Update x-pack/plugins/infra/public/components/metrics_explorer/chart_context_menu.tsx

Co-Authored-By: Felix Stürmer <weltenwort@users.noreply.github.com>

* Adding definite assingment operattors

* Fixing flaky test

Commit:
disable flaky tests

Commit:
[build][quick-fix] don't clean eui (#38483)

clintandrewhall committed:
[Canvas][Easy] Fix EUI Issue in Storybook; standardize locations (#38222)

* Fix Storybook

* Organizing folders a bit

* Update snapshots

Commit:
[Code] change the port number before respawn a new lang-server process (#38090) (#38375)

flash1293 committed:
Use actual listing limit in vis listing (#38246)

ppisljar committed:
Vislib private removal part4 (#37857)

* removing Private

Commit:
disable failing test (#38492)

mattkime committed:
Remove unused injected dependencies 3b (#36899)

* removing more unused injected dependencies

mattkime committed:
remove unused injected dependencies (#38504)

mattkime committed:
remove unused injected dependencies (3rd in a series) (#37947)

* remove unused injected dependencies

mattkime committed:
Remove Private usage from XPackInfoSignatureProvider (#38507)

* remove Private usage, remove mock session storage usage

Commit:
[Code] fix Code multinode mode can't work when security is enabled (#37597)

Commit:
fix(code/frontend): remove load project configs (#38264)

Commit:
fix(code/frontend): should not have empty array as query (#38257)

markov00 committed:
Fix typo in top hit fields filter (#38391)

peteharverson committed:
[ML] Adds check that forecast duration is no more than 3650 days (#38408)

PhilippBaranovskiy committed:
[Accessibility] Reload saved search link on Discover replaced with <EuiButton> component (#37621)

* Reload saved search link replaced with <EuiButton> components: keyboard accessible and looks like a button.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/1189/)
